### DISCUSSION TITLE
Speed up and smooth Animations on Windows

### DISCUSSION THREAD

**Discussion by msenol86** [fltk-rs/fltk-rs#913] [Q&A] [answered]
I have created a card game and animations work super smooth and fluid on Linux. However on Windows animations stutters, I suppose this is related to Windows threading model. 

I have created a [separate thread](https://github.com/msenol86/pistiflex/blob/main/src/main.rs#L71) for animations. I wonder that caused the issue? 

I use [sleep and awake](https://github.com/msenol86/pistiflex/blob/main/src/widget.rs#L103) functions to draw my animation, is there any other way?

https://user-images.githubusercontent.com/2547551/135728539-c90227e1-9190-4b02-a1fa-6572880d6e7e.mp4

.

**Comment by MoAlyousef** [comment]
Nice project.
`app::sleep()` calls `std::thread::sleep()`. Unfortunately on Windows, it has low precision:
https://github.com/alexheretic/spin-sleep#windows-accuracy

You can try the linked crate to see if it helps.
Also see if things improve when building in `release` mode.

**Comment by msenol86** [comment]
I have tried with `release ` but no luck. I will try spin-sleep and update you guys. Thanks 👍

**Comment by msenol86** [comment, marked as answer]
replacing app::sleep with spin_sleep solves the issue. thx for the tip 👍

**Reply by MoAlyousef** [reply]
Glad to hear
